torch/lib/c10d/ProcessGroupGloo.cpp

```diff
@@ -172,14 +172,13 @@ class GlooStore : public ::gloo::rendezvous::Store {
  public:
   GlooStore(const c10::intrusive_ptr<::c10d::Store>& store) : store_(store) {}
 
-  void set(const std::string& key, const std::vector<char>& value) override {
-    std::vector<uint8_t> tmp(value.begin(), value.end());
-    store_->set(key, tmp);
+  void set(const std::string& key, const std::vector<uint8_t>& value) override {
```

Comment:
Let's actually add entirely new set() and get() methods with the desired signatures so we don't break existing use cases and still have the compatibility. The old methods can still stay.